src/main.cpp

```diff
@@ -5292,20 +5322,31 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             }
         }
 
-        if (!vToFetch.empty())
-            pfrom->PushMessage(NetMsgType::GETDATA, vToFetch);
+        // BU remove, unused       if (!vToFetch.empty())
+        // BU remove, unused          pfrom->PushMessage(NetMsgType::GETDATA, vToFetch);
     }
 
 
     else if (strCommand == NetMsgType::GETDATA)
     {
         vector<CInv> vInv;
         vRecv >> vInv;
-        if (vInv.size() > MAX_INV_SZ)
+        if ((vInv.size() > MAX_INV_SZ)||(vInv.size() == 0))  // BU check size == 0 to be intolerant of an empty and useless request
         {
             Misbehaving(pfrom->GetId(), 20);
             return error("message getdata size() = %u", vInv.size());
         }
+        for (unsigned int nInv = 0; nInv < vInv.size(); nInv++)  // Validate that INVs are a valid type
+        {
+            const CInv &inv = vInv[nInv];
+            if (!((inv.type == MSG_TX) || (inv.type == MSG_BLOCK) || (inv.type == MSG_FILTERED_BLOCK) || (inv.type == MSG_THINBLOCK) || (inv.type == MSG_XTHINBLOCK)))
```

Comment:
If you add a new type, we don't want GETDATA to attempt to process your new type because it has no way of doing so.

You would need to extend GETDATA if you want it to service your type.  And in doing so, you can add your type to this validity check.